system-test/datastore.ts

```diff
@@ -348,6 +348,37 @@ async.each(
           await datastore.delete(postKey);
         });
 
+        it('should save a nested name/value pair with name as a long string', async () => {
+          const longString = Buffer.alloc(1501, '.').toString();
+          const postKey = datastore.key(['Post', 'post1']);
+          await datastore.save({
+            key: postKey,
+            data: {
+              metadata: [
+                {
+                  name: longString,
+                  value: 'some-value',
+                },
+              ],
+            },
+            excludeLargeProperties: true,
+          });
+        });
+        it('should save a nested name/value pair with value as a long string', async () => {
```

Comment:
This test shows that even though we are omitting the code that appends `entry.excludeFromIndexes = true;` to the entity that `entry.excludeFromIndexes = true;` is still applied in the right places for entities that are arrays.